### DISCUSSION TITLE
Objects: Multi-select: Insert multiple options at once

### DISCUSSION THREAD

**Discussion by livingstonlarus** [twentyhq/twenty#5539] [Ideas]
In Settings > Data model after selecting an object and then clicking "Add Field", then in section Type and values, if selecting type "Multi-select" then the user needs to input each option one by one with "Add option" button.

This can be tedious if you have multiple multi-select fields to fill with dozens of options.

I suggest that an alternative input mode be offered, like a textarea where the user can type or paste a list of options (one option per line). This is something I've seen in Pipedrive.

**Comment by greptile-apps** [comment]
**Disclaimer:** This comment was automatically generated with AI. It can give some useful hints as to where to look for, but sometimes it can also give very inaccurate answers so takes this with a pinch of salt :).


To implement a textarea for bulk insertion of multi-select options:

1. **Modify `SettingsObjectFieldSelectForm` component**:
   - Add a textarea input for bulk option insertion.
   - Update the `handleSetOptionAsDefault` and `handleRemoveOptionAsDefault` functions to handle multiple options.

2. **Update form rendering**:
   - Refer to `SettingsDataModelFieldSelectSettingsFormCard` for form rendering logic.

3. **Update type selection**:
   - Refer to `SettingsObjectFieldTypeSelectSection` for handling type selection logic.

This will allow users to input multiple options at once for multi-select fields.



### References

**Comment by charlesBochet** [comment]
@Bonapara FYI

**Comment by Bonapara** [comment]
We should add a `⌘⏎` or `⏎` shortcut that adds a new option when typing a value.

**Comment by livingstonlarus** [comment]
@Bonapara interesting, but does not solve the issue. If the user needs to insert 50 options, she still has to do it one by one, even if it's easier with `⏎`.

What's needed is to be able to paste a full list. Either with a textarea, or maybe just capturing the paste event when focus is in the option input, and populate as many new option inputs as needed. That would be even more elegant than a textarea. But it's not obvious, so a tip should make clear that you can paste a list instead of typing it.

**Comment by Bonapara** [comment]
@livingstonlarus, thanks for sharing this use case. We won't implement this in the short term, but we'll keep it in mind and turn this issue into a discussion so people can upvote it!

**Comment by nickwild-999** [comment]
+1 for me on this please

**Comment by BOHEUS** [comment]
Implemented in #17459
